src/ast/semantic_analyser.cpp

```diff
@@ -1240,7 +1240,14 @@ void SemanticAnalyser::visit(AttachPoint &ap)
     auto paths = resolve_binary_path(ap.target);
     if (paths.size() > 1)
       err_ << "path '" << ap.target << "' must refer to a unique binary but matched " << paths.size() << std::endl;
-    ap.target = paths.front();
+
+    if (bpftrace_.pid_ > 0) {
+      std::ostringstream abs;
+      abs << "/proc/" << bpftrace_.pid_ << "/root/" << paths.front();
```

Comment:
should we make a function that does this? Makes it easier to reuse/refactor when needed.

Reply:
Thanks for the suggestion, I imagine this will probably be needed elsewhere at some point.

I put it in util.cpp